tests/unit_tests/encoder/date/test_datetime.py

```diff
@@ -1,30 +1,56 @@
 import unittest
 from datetime import datetime
 import numpy as np
-from dateutil.parser import parse as parse_datetime
+import pandas as pd
 import torch
 from lightwood.encoder.datetime.datetime import DatetimeEncoder
 from lightwood.encoder.datetime.datetime_sin_normalizer import DatetimeNormalizerEncoder
+np.random.seed(1)
 
 
 class TestDatetimeEncoder(unittest.TestCase):
-    def test_decode(self):
-        data = [1555943147, None, 1555943147, '', np.nan]
+    @staticmethod
+    def _create_timestamp():
+        min_ts = pd.Timestamp.min
+        max_ts = pd.Timestamp.max
+        return np.random.randint(min_ts.timestamp(), max_ts.timestamp())
+
+    def test_raise_encode_type(self):
+        data = [1, 2, 3]
+        enc = DatetimeEncoder()
+        enc.prepare([])
+        self.assertRaises(Exception, enc.encode, data)
 
+    def test_encode(self):
+        data = pd.Series([self._create_timestamp() for _ in range(10_000)])
+        data[0] = np.nan
+        enc = DatetimeEncoder()
+        enc.prepare([])
+        encoded_repr = enc.encode(data)
+        assert not torch.isinf(encoded_repr).any()
+        assert not torch.isnan(encoded_repr).any()
+
+    def test_decode(self):
+        data = pd.Series([self._create_timestamp() for _ in range(1_000)])
+        data[0] = np.nan
         enc = DatetimeEncoder()
         enc.prepare([])
         encoded_repr = enc.encode(data)
         assert not torch.isinf(encoded_repr).any()
         assert not torch.isnan(encoded_repr).any()
         dec_data = enc.decode(encoded_repr)
-        for d in dec_data:
-            assert d in data
 
-    def test_sinusoidal_encoding(self):
-        dates = ['1971-12-1 00:01', '2000-5-29 23:59:30', '2262-3-11 3:0:5']
-        dates = [parse_datetime(d) for d in dates]
-        data = [d.timestamp() for d in dates]
+        for d, t in zip(dec_data[1:], data.tolist()[1:]):
+            # ignore edge cases within border of supported years
+            if pd.Timestamp.min.year + 1 < datetime.fromtimestamp(t).year < pd.Timestamp.max.year - 1:
+                if not np.isclose(d, t, atol=5):
+                    assert datetime.fromtimestamp(d) == datetime.fromtimestamp(t)
+                else:
+                    assert np.isclose(d, t, atol=5)
+        assert np.isnan(dec_data[0])
 
+    def test_sinusoidal_encoding(self):
+        data = [self._create_timestamp() for _ in range(100)]
         normalizer = DatetimeNormalizerEncoder(sinusoidal=True)
```

Comment:
I see that there is a trigonometric encoder already! Nice. Probably worth to keep this as a default in Lightwood :D

Reply:
Will add an issue to actually benchmark this and change if the impact is positive (which I agree it _should_)

Reply:
See: #1157